c++/src/LzoDecompressor.cc

```diff
@@ -312,13 +312,11 @@ namespace orc {
               output += SIZE_OF_INT;
               matchAddress += increment32;
 
-              *reinterpret_cast<int32_t*>(output) =
-                *reinterpret_cast<int32_t*>(matchAddress);
+              memcpy(output, matchAddress, SIZE_OF_INT);
```

Comment:
`reinterpret_cast` is a compile-time directive while `memcpy` is a run-time operation. I'm wondering if there is a potential performance regression.